### PR TITLE
[WIP][Jaya | Nidhi] Refactor. and Integrate. TravelClassList

### PR DESCRIPTION
**Pull request by prochetab1999**

### CODE REVIEW

File: src/components/Catalog.js [hunk 1]
```diff
@@ -24,42 +23,41 @@ const providerItems = (
...
-      />
+      <>
+        <Item
+          key={item.id}
+          item={item}
+          onItemSelect={handleItemSelect}
+          isSelected={!item.travelClass && isSelected}
+          isParent={item.travelClass}
+        />
+        {item?.travelClass && (
```
Comment (hariharanweb) [inline src/components/Catalog.js:47]: Item cannot be null

File: src/components/Catalog.js [hunk 2]
```diff
@@ -24,42 +23,41 @@ const providerItems = (
...
+        />
+        )}
+      </>
     );
   });
   return (
     <>
-      {groupedItems}
-      {individualItems}
+      {' '}
```
Comment (hariharanweb) [inline src/components/Catalog.js:59]: Why do you need empty

File: src/components/Item.js [hunk 3]
```diff
@@ -95,46 +96,21 @@ const Item = ({
...
-        </Typography>
-        )}
-        {!isParent && category
-        && (
-        <Typography
-          variant="body1"
-          style={{
-            fontSize: 'small', fontWeight: '600', marginLeft: '10px', paddingTop: '20px', marginTop: '5px',
-          }}
+        <div style={{
```
Comment (hariharanweb) [inline src/components/Item.js:99]: Use grid

File: src/components/Item.js [hunk 4]
```diff
@@ -11,16 +11,17 @@ import './Item.css';
...
-    );
+    if (!item.travelClass) {
+      onItemSelect(
+        item,
+        !isSelected,
+      );
+    }
   };
-  const containerStyle = isParent ? 'parent-item' : 'item-with-border';
+  const containerStyle = 'item-with-border';
```
Comment (hariharanweb) [inline src/components/Item.js:24]: Remove this variable

File: src/components/Item.js [hunk 5]
```diff
@@ -95,46 +96,21 @@ const Item = ({
...
-          </div>
-        )}
+          <Typography variant="body1" style={{ color: 'grey', fontSize: 'small' }}>
+            <DirectionsCarOutlinedIcon style={{
+              fontSize: 'small', marginRight: '2px', color: 'grey',
+            }}
+            />
+            Vehicle
+          </Typography>
+          <Typography variant="body1" style={{ fontSize: 'small', fontWeight: '500', paddingLeft: '1px' }}>
```
Comment (hariharanweb) [inline src/components/Item.js:110]: Use attribute than style.

File: src/components/TravelClass.js [hunk 6]
```diff
@@ -2,25 +2,35 @@ import React from 'react';
...
+      backgroundColor={isTravelClassSelected ? 'rgba(50, 123, 24, 0.1)' : '#EFF6FE'}
+      paddingY="8px"
+      paddingX="8px"
+      justifyContent="space-evenly"
+      flexWrap="nowrap"
+      onClick={onSelectTravelClass}
+    >
+      <Grid item direction="column" textAlign="center" paddingX="8px">
+        <Typography fontWeight={600}>{travelClass?.travel_class_name}</Typography>
+        <Typography fontWeight={500}>{travelClass?.availability}</Typography>
```
Comment (hariharanweb) [inline src/components/TravelClass.js:26]: Handle availability not there

**Review by hariharanweb — CHANGES_REQUESTED** [review]

File: src/components/Catalog.js (hunk 2 above)
Comment (nidhirajput13) [reply src/components/Catalog.js:48]: Fixed

File: src/components/Item.js (hunk 3 above)
Comment (nidhirajput13) [reply src/components/Item.js]: Fixed

File: src/components/TravelClass.js (hunk 6 above)
Comment (nidhirajput13) [reply src/components/TravelClass.js:28]: Working fine

File: src/components/Item.js (hunk 5 above)
Comment (nidhirajput13) [reply src/components/Item.js]: Fixed

**Pull request merged**